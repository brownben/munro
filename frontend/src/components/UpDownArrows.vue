<!--
  Up and Down Arrows for Table

  The arrows for the table indicating sort preference and direction. The data is passed through props and watched for changes.
-->

<template>
  <div class="up-down-arrow">
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      stroke-width="1.5"
      stroke="currentColor"
      fill="none"
      stroke-linecap="round"
      stroke-linejoin="round"
      class="inline-block transition-all duration-300 text-main-900"
      :class="{
        'h-2 -mt-2': !ascending && active,
        'h-5 -mt-1': ascending && active,
        'h-7/2 -mt-1 text-opacity-75': !active,
      }"
    >
      <path stroke="none" d="M0 0h24v24H0z" />
      <line x1="12" y1="5" x2="12" y2="19" />
      <line x1="16" y1="9" x2="12" y2="5" />
      <line x1="8" y1="9" x2="12" y2="5" />
    </svg>
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      stroke-width="1.5"
      stroke="currentColor"
      fill="none"
      stroke-linecap="round"
      stroke-linejoin="round"
      class="inline-block -ml-2 transition-all duration-300 text-main-900"
      :class="{
        'h-2 mt-2': ascending && active,
        'h-5 -mt-1': !ascending && active,
        'h-7/2 -mt-1 text-opacity-75': !active,
      }"
    >
      <path stroke="none" d="M0 0h24v24H0z" />
      <line x1="12" y1="5" x2="12" y2="19" />
      <line x1="16" y1="15" x2="12" y2="19" />
      <line x1="8" y1="15" x2="12" y2="19" />
    </svg>
  </div>
</template>

<script>
export default {
  name: 'UpDownArrows',

  props: {
    active: {
      type: Boolean,
      default: false,
    },

    ascending: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
<style>
.h-7\/2 {
  height: 0.875rem;
}
</style>
