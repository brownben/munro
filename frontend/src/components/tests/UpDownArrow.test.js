/*
  UpDownArrows Component Unit Tests
*/

import { mount } from '@vue/test-utils'
import UpDownArrows from '@/components/UpDownArrows'

test('Is a Vue Instance', () => {
  const wrapper = mount(UpDownArrows)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly - Passed with Props', () => {
  const wrapper = mount(UpDownArrows, {
    propsData: {
      active: true,
      ascending: false,
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Check Classes Update', () => {
  const wrapper = mount(UpDownArrows, {
    propsData: {
      active: true,
      ascending: true,
    },
  })
  expect(wrapper.props().active).toBeTruthy()
  expect(wrapper.props().ascending).toBeTruthy()
  expect(wrapper.findAll('svg').at(0).classes()).toContain('h-5')
  expect(wrapper.findAll('svg').at(1).classes()).toContain('h-2')

  const wrapper1 = mount(UpDownArrows, {
    propsData: {
      active: true,
      ascending: false,
    },
  })
  expect(wrapper1.findAll('svg').at(0).classes()).toContain('h-2')
  expect(wrapper1.findAll('svg').at(1).classes()).toContain('h-5')

  const wrapper2 = mount(UpDownArrows, {
    propsData: {
      active: false,
      ascending: false,
    },
  })
  expect(wrapper2.findAll('svg').at(0).classes()).toContain('h-7/2')
  expect(wrapper2.findAll('svg').at(1).classes()).toContain('h-7/2')
})

test('Renders Correctly - No Props', () => {
  const wrapper = mount(UpDownArrows)
  expect(wrapper.element).toMatchSnapshot()
})
