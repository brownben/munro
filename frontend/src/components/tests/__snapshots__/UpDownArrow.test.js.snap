// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly - No Props 1`] = `
<div
  class="up-down-arrow"
>
  <svg
    class="inline-block transition-all duration-300 text-main-900 h-7/2 -mt-1 text-opacity-75"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      stroke="none"
    />
     
    <line
      x1="12"
      x2="12"
      y1="5"
      y2="19"
    />
     
    <line
      x1="16"
      x2="12"
      y1="9"
      y2="5"
    />
     
    <line
      x1="8"
      x2="12"
      y1="9"
      y2="5"
    />
  </svg>
   
  <svg
    class="inline-block -ml-2 transition-all duration-300 text-main-900 h-7/2 -mt-1 text-opacity-75"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      stroke="none"
    />
     
    <line
      x1="12"
      x2="12"
      y1="5"
      y2="19"
    />
     
    <line
      x1="16"
      x2="12"
      y1="15"
      y2="19"
    />
     
    <line
      x1="8"
      x2="12"
      y1="15"
      y2="19"
    />
  </svg>
</div>
`;

exports[`Renders Correctly - Passed with Props 1`] = `
<div
  class="up-down-arrow"
>
  <svg
    class="inline-block transition-all duration-300 text-main-900 h-2 -mt-2"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      stroke="none"
    />
     
    <line
      x1="12"
      x2="12"
      y1="5"
      y2="19"
    />
     
    <line
      x1="16"
      x2="12"
      y1="9"
      y2="5"
    />
     
    <line
      x1="8"
      x2="12"
      y1="9"
      y2="5"
    />
  </svg>
   
  <svg
    class="inline-block -ml-2 transition-all duration-300 text-main-900 h-5 -mt-1"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      stroke="none"
    />
     
    <line
      x1="12"
      x2="12"
      y1="5"
      y2="19"
    />
     
    <line
      x1="16"
      x2="12"
      y1="15"
      y2="19"
    />
     
    <line
      x1="8"
      x2="12"
      y1="15"
      y2="19"
    />
  </svg>
</div>
`;
